framework/mail/Composer.php

```diff
@@ -0,0 +1,158 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\mail;
+
+use Yii;
+use yii\base\InvalidConfigException;
+use yii\base\Object;
+use yii\web\View;
+
+/**
+ * Composer composes the mail messages via view rendering.
+ *
+ * @property \yii\base\View $view View instance. Note that the type of this property differs in getter and setter. See
+ * [[getView()]] and [[setView()]] for details.
+ * @property string $viewPath The directory that contains the view files for composing mail messages Defaults
+ * to '@app/mail'.
+ *
+ * @author Paul Klimov <klimov.paul@gmail.com>
+ * @since 2.1
+ */
+class Composer extends Object
+{
+    /**
+     * @var string|bool HTML layout view name. This is the layout used to render HTML mail body.
+     * The property can take the following values:
+     *
+     * - a relative view name: a view file relative to [[viewPath]], e.g., 'layouts/html'.
+     * - a path alias: an absolute view file path specified as a path alias, e.g., '@app/mail/html'.
+     * - a boolean false: the layout is disabled.
+     */
+    public $htmlLayout = 'layouts/html';
+    /**
+     * @var string|bool text layout view name. This is the layout used to render TEXT mail body.
+     * Please refer to [[htmlLayout]] for possible values that this property can take.
+     */
+    public $textLayout = 'layouts/text';
+    /**
+     * @var array the configuration that should be applied to any newly created message template.
+     */
+    public $templateConfig = [];
+
+    /**
+     * @var \yii\base\View|array view instance or its array configuration.
+     */
+    private $_view = [];
+    /**
+     * @var string the directory containing view files for composing mail messages.
+     */
+    private $_viewPath;
+
+
+    /**
+     * @return string the directory that contains the view files for composing mail messages
+     * Defaults to '@app/mail'.
+     */
+    public function getViewPath()
+    {
+        if ($this->_viewPath === null) {
+            $this->setViewPath('@app/mail');
+        }
+        return $this->_viewPath;
+    }
+
+    /**
+     * @param string $path the directory that contains the view files for composing mail messages
+     * This can be specified as an absolute path or a path alias.
+     */
+    public function setViewPath($path)
+    {
+        $this->_viewPath = Yii::getAlias($path);
+    }
+
+    /**
+     * @param array|\yii\base\View $view view instance or its array configuration that will be used to
+     * render message bodies.
+     * @throws InvalidConfigException on invalid argument.
+     */
+    public function setView($view)
+    {
+        if (!is_array($view) && !is_object($view)) {
+            throw new InvalidConfigException('"' . get_class($this) . '::view" should be either object or configuration array, "' . gettype($view) . '" given.');
```

Comment:
Disagreed: I can see no problem in this - it does not violate any code style rules.
Method `setView()` has been just moved from `BaseMailer` as it was before.